packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -689,7 +805,29 @@ impl<'content> AssetRouter<'content> {
             headers.push(("content-encoding".to_string(), encoding.to_string()));
         }
 
-        Self::prepare_response_and_certification(asset.url.to_string(), 200, asset.content, headers)
+        if let Some(range_begin) = range_begin {
```

Comment:
I like the changes that you've made to the function 👍 

The point for me is that it's actually the caller that decides whether we prepare a range response or a standard response. The `prepare_static_asset` is always called with a `Some` or `None` literal, and never with an unknown `Option` value.

If we hide this logic by putting the condition deep inside the function, it makes the code more difficult to follow in my opinion. 

Since the caller already knows whether it needs a range response or a standard response, then why not directly call a function that is purpose built? And then as a reader you will know instantly what is happening, instead of having to trace the code down two levels of functions to find the condition? This would also remove the need for the conditional check in the first place.

Reply:
Happy to chat more about this if you like, but I don't think it should block the PR so I'll approve. Thanks for the discussion

Reply:
Ok, I'll merge then, but let's chat about potential refactoring. Thanks!